Packs/Uptycs/Integrations/Uptycs/Uptycs.py

```diff
@@ -367,12 +451,166 @@ def uptycs_get_assets_command():
+                               'createdAt', 'lastOccurredAt']
+
+    if context is not None:
+        remove_context_entries(context, context_entries_to_keep)
+
+    human_readable = tableToMarkdown('Uptycs Detections: ',
+                                     context,
+                                     context_entries_to_keep)
+    entry = {
+        'ContentsFormat': formats['json'],
+        'Type': entryTypes['note'],
+        'Contents': query_results,
+        'HumanReadable': human_readable,
+        'EntryContext': {
+            'Uptycs.Detections(val.id == obj.id)': context
+        }
+    }
+
+    return entry
+
+
+def uptycs_get_detection_details():
+    """
+    return detection details
+    """
+    http_method = 'get'
+    api_call = "/detections/"
+    detection_id = demisto.args().get('detection_id')
+
+    if detection_id != -1 and detection_id is not None:
+        api_call = ("%s%s" % (api_call, detection_id))
+
+    return restcall(http_method, api_call)
+
+
+def uptycs_get_detection_details_command():
+    query_results = uptycs_get_detection_details()
+    detection = query_results.get('detection')
+    final_alerts = []
+    detection_url = 'https://' + DOMAIN + '/ui/detections/' + demisto.args().get('detection_id')
+    display_name = get_href_link(detection["displayName"], detection_url)
+
+    for alert in query_results.get('alerts').get('items'):
+        alert_text = alert['alertTime'] + '<br>Alert - ' + alert['alertRuleName']
+        alert_text += '<br>\tscore - ' + alert['score'] + '<br>\t'
+        alert_text += alert['key'] + ' : ' + alert['value'] + '<br>\t'
+        alert_text += get_href_link('Go to Alert details', 'https://' + DOMAIN + '/ui/alerts/' + alert['id'])
+        final_alerts.append(alert_text + '<br>')
+
+    for event in query_results.get('events').get('items'):
+        event_text = event['event_time'] + '\nEvent - '
+        if 'eventRule' in event:
+            event_text += event['eventRule']['name']
+        elif 'event_name' in event:
+            event_text += event['event_name']
+        if 'score' in event:
+            event_text += '\n\tscore - ' + str(event['score'])
+        elif 'severity' in event:
+            event_text += '\n\tseverity - ' + event['severity']
+        if event.get('key') and event.get('value'):
+            event_text += '\n\t' + event['key'] + ' : ' + event['value']
+        final_alerts.append(event_text + '\n')
+
+    context = {'id': detection['id'],
+               'displayName': display_name,
+               'score': detection['score'],
+               'severity': detection['severity'],
+               'createdAt': detection['createdAt'],
+               'signals': detection['signals'],
+               'tacticCount': detection['tacticCount'],
+               'tactics': str(detection['tactics']),
+               'assetHostName': detection['assetHostName'],
+               'Alerts and Events': '<br>'.join(sorted(final_alerts)),
```

Comment:
why do we want to retrieve all the text in the context?
context is better for objects where we can retrieve the information from and move with it to the next task.
I think it's better to create a list of dicts with the info and use the TableToMarkdown.

Reply:
removed redundant statements and updated code to use context properly.